lib/ddtrace/contrib/rack/rum_injection.rb

```diff
@@ -0,0 +1,239 @@
+require 'ddtrace/contrib/rack/ext'
+require 'ddtrace/environment'
+require 'delegate'
+require 'date'
+
+module Datadog
+  module Contrib
+    # Rack module includes middlewares that are required to trace any framework
+    # and application built on top of Rack.
+    module Rack
+      # RumInjection ensures that the Rack Response has rum information
+      # injected into it's html. The middleware modifies the response body
+      # of non-cached html so that it can be retrieved by the rum browser-sdk in
+      # the application frontend.
+      # rubocop:disable Metrics/ClassLength
+      class RumInjection
+        include Datadog::Environment::Helpers
+
+        RUM_INJECTION_FLAG = 'datadog.rum_injection_flag'.freeze
+        INLINE = 'inline'.freeze
+        IDENTITY = 'identity'.freeze
+        HTML_CONTENT = 'text/html'.freeze
+        XHTML_CONTENT = 'application/xhtml+xml'.freeze
+        HEAD_TAG_OPEN = '<head'.freeze
+        TAG_CLOSE = '>'.freeze
```

Comment:
I noticed some of these are not needed anymore.

Reply:
👍  removed